File: src/main/java/javabot/operations/LeaveOperation.java
package javabot.operations;

import com.antwerkz.sofia.Sofia;
import javabot.Message;
import org.pircbotx.Channel;
import org.pircbotx.User;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class LeaveOperation extends BotOperation {
    private static final Logger log = LoggerFactory.getLogger(LeaveOperation.class);

    @Override
    public boolean handleMessage(final Message event) {
        final String message = event.getValue();
        final Channel channel = event.getChannel();
        final User sender = event.getUser();
        if ("leave".equals(message.toLowerCase())) {
            if (channel.getName().equalsIgnoreCase(event.getUser().getNick())) {
                getBot().postMessageToChannel(event, Sofia.leavePrivmsg(sender.getNick()));
            } else {
                getBot().postMessageToChannel(event, Sofia.leaveChannel(event.getUser().getNick()));
                new Thread(() -> {
                    //                        ircBot.getUserChannelDao().getChannel(channel.getName());
                    try {
                        Thread.sleep(60000 * 15);
                    } catch (InterruptedException exception) {
                        log.error(exception.getMessage(), exception);
                    }
                    //                        getBot().getPircBot().joinChannel(channel);
                }).start();
            }
        }
        return false;
    }
}

File: src/main/kotlin/javabot/operations/LeaveOperation.kt
package javabot.operations

import com.antwerkz.sofia.Sofia
import com.google.inject.Inject
import java.util.Locale
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao

class LeaveOperation @Inject constructor(bot: Javabot, adminDao: AdminDao) :
    BotOperation(bot, adminDao) {

    override fun handleMessage(event: Message): List<Message> {
        val responses = arrayListOf<Message>()
        val message = event.value
        val channel = event.channel
        val sender = event.user
        if ("leave" == message.lowercase(Locale.getDefault())) {
            if (channel!!.name.equals(event.user.nick, ignoreCase = true)) {
                responses.add(Message(event, Sofia.leavePrivmsg(sender.nick)))
            } else {
                responses.add(Message(event, Sofia.leaveChannel(event.user.nick)))
                //                        ircBot.getUserChannelDao().getChannel(channel.getName());
                //                        getBot().getPircBot().joinChannel(channel);
            }
        }
        return responses
    }
}
